fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -212,6 +212,33 @@ where
             PowerUpdates::default()
         };
 
+        // check for upgrades in the upgrade_scheduler
+        match &self.upgrade_scheduler {
+            Some(upgrade_scheduler) => {
+                let chain_id: u64 = state.chain_id().into();
+                let block_height: u64 = state.block_height().try_into().unwrap();
+
+                if let Some(upgrade) = upgrade_scheduler.get(chain_id, block_height) {
+                    tracing::info!(
+                        chain_id = chain_id,
+                        height = block_height,
+                        "Running migration",
+                    );
+
+                    // there is an upgrade scheduled for this height, lets run the migration
+                    let res = (upgrade.migration)(&mut state);
```

Comment:
I'm not sure what exactly `res` is here, but if it can fail, it should stop the execution.

Reply:
done